frontend/src/features/public-form/components/FormPaymentPage/stripe/components/DownloadReceiptBlock.tsx

```diff
@@ -1,46 +1,116 @@
+import { useMemo } from 'react'
 import { BiDownload } from 'react-icons/bi'
+import { Box, Divider, Stack, Text } from '@chakra-ui/react'
+import { format } from 'date-fns'
+
+import { PaymentType, ProductItem } from '~shared/types'
+import { centsToDollars } from '~shared/utils/payments'
 
 import { useToast } from '~hooks/useToast'
 import Button from '~components/Button'
 
 import { getPaymentInvoiceDownloadUrl } from '~features/public-form/utils/urls'
 
-import { GenericMessageBlock } from './GenericMessageBlock'
-
 type DownloadReceiptBlockProps = {
   formId: string
   submissionId: string
   paymentId: string
+  amount: number
+  products: ProductItem[]
+  paymentType?: PaymentType
+  name: string
+  paymentDate?: Date
+}
+
+const PaymentSummaryItem = ({
```

Comment:
Since this might not always represent an `Item` (as in product item in the payment/product domain), it might make sense to name this as `PaymentSummaryRow`. wdyt

Reply:
makes sense! [2353977](https://github.com/opengovsg/FormSG/pull/6591/commits/23539777cf1a763a9cb3d35523dc85d0063015f7)